packages/flame/lib/src/collisions/collision_callbacks.dart

```diff
@@ -114,6 +127,22 @@ mixin CollisionCallbacks on Component
     onCollisionEndCallback?.call(other);
   }
 
+  /// Works only for the QuadTree collision detection.
+  /// If you need to prevent collision of items of different types -
+  /// reimplement [broadPhaseCheck]. The result of calculation is cached so you
+  /// should not check any dynamical parameters here, the function intended to
+  /// be used as pure type checker.
+  @mustCallSuper
+  bool broadPhaseCheck(PositionComponent other) {
```

Comment:
I'm not too keen on having this in the callbacks mixin, also since it doesn't work for all systems.

Reply:
But it is actually callback like other collision callbacks are. It is intended to be reimplemented by framework's user. 
And also, the "sweep" algorithm also could be extended by this functionality, but patching original algorithm is out of the scope of this PR, I simply did not analyze it's code. To perform a basic type-checking at broadphase is basically better approach then to perform it after expensive collision check.

Reply:
Hmm, I guess it could be seen as a callback. But it should be named something better then, because the user shouldn't even necessarily have to know what a broadphase is.
I can try to think of something tomorrow if you don't come up with something. :)

Reply:
I agree, something like "onComponentsTypeCheck" would be much better than current naming. But I'm open for propositions of course 👌